CMakeLists.txt

```
@@ -774,45 +739,13 @@ endif()
 # assume no community board... until proven otherwise
 set(IS_COMMUNITY_TARGET FALSE CACHE INTERNAL "Community target flag")
 
-#######################
-# ChibiOS
-if(RTOS_CHIBIOS_CHECK)
-
-    # Define base path for the class libraries
-    nf_set_base_path_for_libraries_modules(${CMAKE_SOURCE_DIR}/targets/ChibiOS/_nanoCLR)
-    
-    add_subdirectory(targets/ChibiOS)
-
-    # now add the subdirectory for the board
-    # try to find board in the targets folder
-    if(EXISTS ${CMAKE_SOURCE_DIR}/targets/ChibiOS/${TARGET_BOARD})
-
-        # board found
-        message(STATUS "Support for target board '${TARGET_BOARD}' found")
-
-        # add TARGET board directory
-        add_subdirectory(${CMAKE_SOURCE_DIR}/targets/ChibiOS/${TARGET_BOARD})
-
-    else()
-
-        # try to find board in the Community targets folder
-        if(EXISTS ${CMAKE_SOURCE_DIR}/targets-community/ChibiOS/${TARGET_BOARD})
-            # board found
-            message(STATUS "Support for target board '${TARGET_BOARD}' found in Community targets")
 
-            # add TARGET board directory from Community
-            add_subdirectory(targets-community/ChibiOS/${TARGET_BOARD})
-        else()
-        # board NOT found in targets folder
-            # board NOT found in targets folder
-            message(FATAL_ERROR "\n\nSorry but support for ${TARGET_BOARD} target is not available...\n\nYou can wait for that to be added or you might want to contribute and start working on a PR for that.\n\n")
-        endif()
-
-    endif()
+# Define PLATFORM base path
+set(BASE_PATH_FOR_PLATFORM ${CMAKE_CURRENT_SOURCE_DIR})
 
 #######################
 # FreeRTOS
-elseif(RTOS_FREERTOS_CHECK)
+if(RTOS_FREERTOS_CHECK)
```

I guess this should be:
```suggestion
if(RTOS_FREERTOS_ESP32_CHECK)
```

---

There are two platforms at the start of the "custom" platforms logic related to FreeRTOS. Please see a couple of lines below your suggested edit the if(RTOS_FREERTOS_ESP32_CHECK)

```
#######################
# FreeRTOS
if(RTOS_FREERTOS_CHECK)

    add_subdirectory(targets/FreeRTOS)

#######################
# FreeRTOS ESP32
elseif(RTOS_FREERTOS_ESP32_CHECK)
```

---

If this is the IF for FREERTOS, then the code below doesn't belong here as it's validating RTC for ESP32, which would belong to _another_ IF....

---

Forget my comment above. I was mislead by the partial representatiuon of the diff on git. After looking at the complete file, I can see the code block is correct.